edgexfoundry/device-sdk-go
# feat: Location of client service obtained from the registry

closes #1091

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) ** Existing unit tests suffice**

## Testing Instructions
Use this branch in Device Virtual
Build Device Virtual
Run non-secure EdgeX stack in docker (stop docker Device Virtual)
Run Device Virtual from command line w/o options with `EDGEX_SECURITY_SECRET_STORE=false` and `WRITABLE_LOGLEVEL=DEBUG`
Verify logs contain the following:
```
level=DEBUG ts=2022-02-02T23:55:23.5182364Z app=device-virtual source=clients.go:124 msg="Using configuration for URL for 'core-metadata': http://loca
lhost:59881"
level=DEBUG ts=2022-02-02T23:55:23.5182852Z app=device-virtual source=clients.go:124 msg="Using configuration for URL for 'core-data': http://localhos
t:59880"
```
Verify logs show existing Device Service and Devices where found in Core Metadata (from Docker version before stopped)
Rerun Device Virtual from command line with `-r` and with `EDGEX_SECURITY_SECRET_STORE=false` and `WRITABLE_LOGLEVEL=DEBUG`
Verify logs contain the following:
```
level=DEBUG ts=2022-02-02T23:55:23.5182364Z app=device-virtual source=clients.go:124 msg="Using configuration for URL for 'core-metadata': http://loca
lhost:59881"
level=DEBUG ts=2022-02-02T23:55:23.5182852Z app=device-virtual source=clients.go:124 msg="Using configuration for URL for 'core-data': http://localhos
t:59880"
```
level=DEBUG ts=2022-02-02T23:58:57.2584294Z app=device-virtual source=clients.go:146 msg="Using registry for URL for 'core-data': http://edgex-core-da
ta:59880"
level=DEBUG ts=2022-02-02T23:58:57.2596585Z app=device-virtual source=clients.go:146 msg="Using registry for URL for 'core-metadata': http://edgex-cor
e-metadata:59881"
```
Verify the service got the following error and stopped beacus it can't reach `edgex-core-metadata:59881` which is only on the Docker network.
level=ERROR ts=2022-02-02T23:58:57.295353Z app=device-virtual source=init.go:42 msg="Failed to init cache: failed to send a http request -> Get \"http
://edgex-core-metadata:59881/api/v2/device/service/name/device-virtual?limit=-1&offset=0\": dial tcp: lookup edgex-core-metadata on 8.8.8.8:53: no suc
h host"

## New Dependency Instructions (If applicable)
**N/A**
